resources/fetchers/process_fetcher.go

```diff
@@ -19,36 +32,99 @@ type ProcessesFetcher struct {
 
 type ProcessFetcherConfig struct {
 	fetching.BaseFetcherConfig
-	Directory string `config:"directory"` // parent directory of target procfs
+	Directory         string `config:"directory"` // parent directory of target procfs
+	Fs                fs.FS
+	RequiredProcesses ProcessesConfigMap `config:"processes"`
 }
 
 func (f *ProcessesFetcher) Fetch(ctx context.Context) ([]fetching.Resource, error) {
-	pids, err := proc.List(f.cfg.Directory)
+	pids, err := proc.ListFS(f.cfg.Fs)
 	if err != nil {
 		return nil, err
 	}
-
 	ret := make([]fetching.Resource, 0)
 
 	// If errors occur during read, then return what we have till now
 	// without reporting errors.
 	for _, p := range pids {
-		cmd, err := proc.ReadCmdLine(f.cfg.Directory, p)
+		stat, err := proc.ReadStatFS(f.cfg.Fs, p)
 		if err != nil {
-			return ret, nil
+			return nil, err
+		}
+		processConfig, isProcessRequired := f.cfg.RequiredProcesses[stat.Name]
+		if !isProcessRequired {
+			continue
 		}
 
-		stat, err := proc.ReadStat(f.cfg.Directory, p)
+		fetchedResource, err := f.fetchProcessData(stat, processConfig, p)
 		if err != nil {
-			return ret, nil
+			logp.Error(fmt.Errorf("%+v", err))
+			continue
 		}
-
-		ret = append(ret, ProcessResource{PID: p, Cmd: cmd, Stat: stat})
+		ret = append(ret, fetchedResource)
 	}
 
 	return ret, nil
 }
 
+func (f *ProcessesFetcher) fetchProcessData(procStat proc.ProcStat, processConf ProcessInputConfiguration, processId string) (fetching.Resource, error) {
+	cmd, err := proc.ReadCmdLineFS(f.cfg.Fs, processId)
+	if err != nil {
+		return nil, err
+	}
+	configMap := f.getProcessConfigurationFile(processConf, cmd, procStat.Name)
+
+	return ProcessResource{PID: processId, Cmd: cmd, Stat: procStat, ExternalData: configMap}, nil
+}
+
+//getProcessConfigurationFile - reads the configuration file associated with a process.
+// As an input this function receives a ProcessInputConfiguration that contains CommandArguments, a string array that represents some process flags
+// The function extracts the configuration file associated with each flag and returns it.
+func (f *ProcessesFetcher) getProcessConfigurationFile(processConfig ProcessInputConfiguration, cmd string, processName string) map[string]interface{} {
+	configMap := make(map[string]interface{}, 0)
+	for _, argument := range processConfig.CommandArguments {
+		// The regex extracts the cmd line flag(argument) value
+		regex := fmt.Sprintf(CMDArgumentMatcher, argument)
+		matcher := regexp.MustCompile(regex)
+		if !matcher.MatchString(cmd) {
+			logp.Error(fmt.Errorf("failed to find argument %s in process %s", argument, processName))
```

Comment:
Why do you think that we shouldn't add an error here ?
According to the config you want to get a certain config argument for a certain process, I think that debug wise it's an error that should be printed (I don't find it any different from printing error when the config search for a file that doesn't exist).

As for the success log - I don't mind adding it, but looking at the other fetchers in case I was able to fetch file/kube/process I don't log anything, so what is the difference here. I get your point, but I think that it can cause a massive amount of logs, if we will logs all these small events.

Reply:
1. Regarding the case that we couldn't find the regex, as we both know, a configuration file is just one way of configuring a process, another way will be using command line arguments. In case our user uses cli args I don't see a point in endlessly printing error logs.
2. Regarding the success flow, I just want us to be able somehow to distinguish between the multiple running modes that the process fetcher has: with or without the configuration file. So the two info logs should cover that requirement.